packages/netlify-cms-backend-bitbucket/src/implementation.js

```diff
@@ -225,36 +227,70 @@ export default class BitbucketBackend {
     }));
   }
 
-  getMedia() {
+  getMedia(mediaFolder = this.config.get('media_folder')) {
     return this.api
-      .listAllFiles(this.config.get('media_folder'))
+      .listAllFiles(mediaFolder)
       .then(files =>
         files.map(({ id, name, path }) => ({ id, name, path, displayURL: { id, path } })),
       );
   }
 
+  getMediaAsBlob(path, id) {
+    return this.api.readFile(path, id, { parseText: false });
+  }
+
   getMediaDisplayURL(displayURL) {
     this._mediaDisplayURLSem = this._mediaDisplayURLSem || semaphore(MAX_CONCURRENT_DOWNLOADS);
     const { id, path } = displayURL;
     return new Promise((resolve, reject) =>
       this._mediaDisplayURLSem.take(() =>
-        this.api
-          .readFile(path, id, { parseText: false })
+        this.getMediaAsBlob(path, id)
           .then(blob => URL.createObjectURL(blob))
           .then(resolve, reject)
           .finally(() => this._mediaDisplayURLSem.leave()),
       ),
     );
   }
 
+  async getMediaFile(path) {
```

Comment:
Support loading a specific media file by path.
Usually the redux store will have all the relevant media files loaded (when opening the media library), but when switching to collections grid view, we need to display the cover image for an entry that hasn't been loaded fully (thus its media files are not in the redux store).

Reply:
Good catch